forms/session/authentication_form_test.go

```diff
@@ -0,0 +1,246 @@
+package forms_test
+
+import (
+	"github.com/beego/beego/v2/core/validation"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	form "go-crawler-challenge/forms/session"
+	. "go-crawler-challenge/tests"
+	. "go-crawler-challenge/tests/fixtures"
+	"strings"
+)
+
+var _ = Describe("Session/AuthenticationForm", func() {
+	AfterEach(func() {
+		TruncateTable("user")
+	})
+
+	Describe("#Valid", func() {
+		Context("given valid params", func() {
+			It("does NOT produces any errors", func() {
+				_ = FabricateUser("dev@nimblehq.co", "password")
+				form := form.AuthenticationForm{
+					Email:    "dev@nimblehq.co",
+					Password: "password",
+				}
+
+				formValidation := validation.Validation{}
+				form.Valid(&formValidation)
+
+				Expect(len(formValidation.Errors)).To(BeZero())
+			})
+		})
+
+		Context("given INVALID params", func() {
+			Context("given the email does NOT exist", func() {
+				It("produces an error", func() {
+					form := form.AuthenticationForm{
+						Email:    "dev@nimblehq.co",
+						Password: "password",
+					}
+
+					formValidation := validation.Validation{}
+					form.Valid(&formValidation)
+
+					Expect(len(formValidation.Errors)).To(Equal(1))
+					Expect(formValidation.Errors[0].Key).To(Equal("Email"))
+					Expect(formValidation.Errors[0].Message).To(Equal("Your email or password is incorrect"))
+				})
+			})
+
+			Context("given an mismatch password", func() {
+				It("produces an error", func() {
+					_ = FabricateUser("dev@nimblehq.co", "password")
+					form := form.AuthenticationForm{
+						Email:    "dev@nimblehq.co",
+						Password: "INVALID_PASSWORD",
+					}
+
+					formValidation := validation.Validation{}
+					form.Valid(&formValidation)
+
+					Expect(len(formValidation.Errors)).To(Equal(1))
+					Expect(formValidation.Errors[0].Key).To(Equal("Email"))
+					Expect(formValidation.Errors[0].Message).To(Equal("Your email or password is incorrect"))
+				})
+			})
+		})
+	})
+
+	Describe("#Authenticate", func() {
+		Context("given valid params", func() {
+			It("returns user object", func() {
+				user := FabricateUser("dev@nimblehq.co", "password")
+				form := form.AuthenticationForm{
+					Email:    "dev@nimblehq.co",
+					Password: "password",
+				}
+
+				currentUser, _ := form.Authenticate()
+
+				Expect(currentUser).NotTo(BeNil())
+				Expect(currentUser.Id).To(Equal(user.Id))
+			})
+
+			It("does NOT produce any errors", func() {
+				_ = FabricateUser("dev@nimblehq.co", "password")
+				form := form.AuthenticationForm{
+					Email:    "dev@nimblehq.co",
+					Password: "password",
+				}
+
+				_, errors := form.Authenticate()
+
+				Expect(len(errors)).To(BeZero())
+			})
+		})
+
+		Context("given INVALID params", func() {
+			Context("given the email does NOT exist", func() {
+				It("does NOT return a user object", func() {
```

Comment:
```suggestion
				It("does NOT return an user object", func() {
```

Reply:
@junan I think we should use `a user` regarding [this](https://www.quora.com/Which-is-the-correct-usage-a-user-or-an-user#:~:text=The%20correct%20usage%20would%20be,uniform%2C%20a%20ewe%2C%20etc.) because `user` pronounce with `y` which not a vowel sound

Reply:
@carryall My bad! @Lahphim Please ignore my comments related to it.